crates/rpc/rpc/src/admin.rs

```diff
@@ -115,27 +115,67 @@ where
             ..self.chain_spec.genesis().config.clone()
         };
 
-        let node_info = NodeInfo {
-            id: B256::from_slice(&enode.id.as_slice()[..32]),
-            name: status.client_version,
-            enode: enode.to_string(),
-            enr: self.network.local_enr().to_string(),
-            ip: enode.address,
-            ports: Ports { discovery: enode.udp_port, listener: enode.tcp_port },
-            listen_addr: enode.tcp_addr(),
-            protocols: ProtocolInfo {
-                eth: Some(EthProtocolInfo {
-                    network: status.eth_protocol_info.network,
-                    difficulty: status.eth_protocol_info.difficulty,
-                    genesis: status.eth_protocol_info.genesis,
-                    config,
-                    head: status.eth_protocol_info.head,
-                }),
-                snap: None,
-            },
-        };
+        config.homestead_block =
+            self.chain_spec.hardforks.fork_block_number_or_time(EthereumHardfork::Homestead);
+        config.dao_fork_block =
+            self.chain_spec.hardforks.fork_block_number_or_time(EthereumHardfork::Dao);
+        config.eip150_block =
+            self.chain_spec.hardforks.fork_block_number_or_time(EthereumHardfork::Tangerine);
+        config.eip155_block =
```

Comment:
maybe we can bake this into the Chainspec type somehow,

but I'd like if we solve this with a macro, that does all of this for `(field = Hardfork)*`